generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java

```diff
@@ -316,7 +316,10 @@ private String fieldAbstractJavaType(FieldSpec field) {
             return "UUID";
         } else if (field.type().isString()) {
             return "String";
-        } else if (field.type().isBytes()) {
+        } else if (field.type().isBytes() && field.useByteBuffer()) {
```

Comment:
nit: we could have the nested condition within `field.type().isBytes()`

Reply:
fixed.